hive-1-bigquery-connector/src/main/java/com/google/cloud/hive/bigquery/connector/output/PreInsertHook.java

```diff
@@ -70,7 +70,7 @@ public void run(HookContext hookContext) throws Exception {
       context.setCmd(hookContext.getQueryPlan().getQueryString());
       ParseDriver parseDriver = new ParseDriver();
       ASTNode tree = parseDriver.parse(hookContext.getQueryPlan().getQueryString(), context);
-      HiveConf hiveConf = new HiveConf(conf, HiveConf.class);
+      HiveConf hiveConf = new HiveConf(conf, this.getClass());
```

Comment:
i missed this part before. are we using analyzer to just find out if it is overwrite? seems very heavy handed. is the info not available in HookContext's queryplan?

Reply:
We have to do this because unfortunately the WriteEntity objects in hookContext.getOutputs() are systematically marked as being of type INSERT_OVERWRITE, regardless of whether it is an "INSERT OVERWRITE" query or a regular "INSERT" query.
I've added a comment in the code to clarify.

Reply:
Also to clarify, this hook is only used in Hive 1.x.x.